samples/sensor/stm32_temp_sensor/src/main.c

```diff
@@ -15,6 +15,8 @@ void main(void)
 	const struct device *const dev = DEVICE_DT_GET_ONE(st_stm32_temp);
 #elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_temp_cal)
 	const struct device *const dev = DEVICE_DT_GET_ONE(st_stm32_temp_cal);
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_temp_cal_c0)
+	const struct device *const dev = DEVICE_DT_GET_ONE(st_stm32_temp_cal_c0);
```

Comment:
Can we merge this sample with the more generic `samples/sensor/die_temp_polling`